pan-domain-auth-verification/src/main/scala/com/gu/pandomainauth/Settings.scala

```diff
@@ -0,0 +1,66 @@
+package com.gu.pandomainauth
+
+import java.io.ByteArrayInputStream
+import java.util.Properties
+
+import com.amazonaws.services.s3.AmazonS3
+import com.amazonaws.util.IOUtils
+import org.slf4j.Logger
+
+import scala.util.control.NonFatal
+import scala.collection.JavaConverters._
+
+sealed trait SettingsFailure
+case class SettingsDownloadFailure(cause: Throwable) extends SettingsFailure
+case class SettingsParseFailure(cause: Throwable) extends SettingsFailure
+case object PublicKeyFormatFailure extends SettingsFailure
+case object PublicKeyNotFoundFailure extends SettingsFailure
+
+object Settings {
+  // internal functions for fetching and parsing the responses
+  def fetchSettings(settingsFileKey: String, bucketName: String, s3Client: AmazonS3): Either[SettingsFailure, String] = try {
```

Comment:
Both the normal settings refresher and the public settings refresher parsed their settings in very similar ways so I have refactored them to use the same code here.

I have also changed to using an `Either` type rather than exceptions. The existing behaviour for each settings refresher is maintained though. The public settings refresher never throws an exception whereas the normal refresher throws an exception (created by wrapping the Failure) if the settings cannot be loaded at construction.